tests/pykafka/test_simpleconsumer.py

```diff
@@ -55,18 +65,46 @@ def test_offset_resume(self):
         with self._get_simple_consumer(
                 consumer_group='test_offset_resume') as consumer:
             [consumer.consume() for _ in xrange(100)]
-            offsets_committed = self._currently_held_offsets(consumer)
+            offsets_committed = consumer.held_offsets
             consumer.commit_offsets()
 
         with self._get_simple_consumer(
                 consumer_group='test_offset_resume') as consumer:
-            offsets_resumed = self._currently_held_offsets(consumer)
-            self.assertEquals(offsets_resumed, offsets_committed)
+            self.assertEquals(consumer.held_offsets, offsets_committed)
 
-    @staticmethod
-    def _currently_held_offsets(consumer):
-        return dict((p.partition.id, p.last_offset_consumed)
-                    for p in consumer._partitions.itervalues())
+    def test_reset_offset_on_start(self):
+        """Try starting from LATEST and EARLIEST offsets"""
+        with self._get_simple_consumer(
+                auto_offset_reset=OffsetType.EARLIEST,
+                reset_offset_on_start=True) as consumer:
+            earliest_offs = consumer.held_offsets
+            self.assertIsNotNone(consumer.consume())
+        with self._get_simple_consumer(
+                auto_offset_reset=OffsetType.LATEST,
+                reset_offset_on_start=True,
+                consumer_timeout_ms=500) as consumer:
+            latest_offs = consumer.held_offsets
+            self.assertIsNone(consumer.consume(block=False))
+        difference = sum(
+            latest_offs[i] - earliest_offs[i] for i in latest_offs.keys())
+        self.assertEqual(difference, self.total_msgs)
+
+    def test_reset_offsets(self):
+        """Test resetting to user-provided offsets"""
+        with self._get_simple_consumer(
+                auto_offset_reset=OffsetType.LATEST,
+                reset_offset_on_start=True) as consumer:
+            latest_offs = consumer.held_offsets
+            # We can't control which partition(s) our setUpClass produce calls
+            # write to, so we don't know on which partitions the following will
+            # result in a valid offset. But at least one will be valid, which
+            # is good enough.
+            custom_offs = [(consumer.partitions[pid], latest - 5)
+                           for pid, latest in latest_offs.items()]
+            consumer.reset_offsets(partition_offsets=custom_offs)
+            msg = consumer.consume()
+            # FIXME this 4 should be a 5 or I'm getting something else wrong:
+            self.assertEqual(msg.offset, latest_offs[msg.partition_id] - 4)
```

Comment:
Ah - the problem is in the test assertion of course: we're comparing a `last_offset_consumed` to the next `msg.offset`, so these _should_ be off by one. Sorry for the noise.